File: backend/insurance_engine/src/InsuranceEngine.SharedKernel/Domain/ValueObjects/Address.cs
namespace InsuranceEngine.SharedKernel.Domain.ValueObjects;

/// <summary>
/// Value object for address information.
/// </summary>
public record Address
{
    public string? AddressLine1 { get; init; }
    public string? AddressLine2 { get; init; }
    public string? City { get; init; }
    public string? District { get; init; }
    public string? Division { get; init; }
    public string? PostalCode { get; init; }
    public string Country { get; init; } = "Bangladesh";
    public double? Latitude { get; init; }
    public double? Longitude { get; init; }

    public Address() { }

    public Address(string addressLine1, string city, string? district = null, string country = "Bangladesh")
    {
        AddressLine1 = addressLine1;
        City = city;
        District = district;
        Country = country;
    }
}

File: sdks/insuretech-go-sdk/pkg/models/address.go
package models


// Address represents a address
type Address struct {
	AddressLine1 string `json:"address_line1,omitempty"`
	AddressLine2 string `json:"address_line2,omitempty"`
	City string `json:"city,omitempty"`
	Country string `json:"country,omitempty"`
	District string `json:"district,omitempty"`
	Division string `json:"division,omitempty"`
	Latitude float64 `json:"latitude,omitempty"`
	Longitude float64 `json:"longitude,omitempty"`
	PostalCode string `json:"postal_code,omitempty"`
}
